catalog/dags/common/extensions.py

```diff
@@ -1,12 +1,44 @@
-EXTENSIONS = {
-    "image": {"jpg", "jpeg", "png", "gif", "bmp", "webp", "tiff", "tif", "svg"},
-    "audio": {"mp3", "ogg", "wav", "aiff", "flac", "wma", "mp4", "aac", "m4a", "m4b"},
+import mimetypes
+
+from common.constants import MEDIA_TYPES as SUPPORTED_TYPES
+
+
+FILETYPE_EQUIVALENTS = {
+    # Image extensions
+    "jpeg": "jpg",
+    "tif": "tiff",
+    # Audio extensions
+    "midi": "mid",
 }
 
+mimetypes.add_type("audio/midi", ".mid")
+mimetypes.add_type("audio/midi", ".midi")
+mimetypes.add_type("audio/x-matroska", ".mka")
+
+
+class InvalidFiletypeError(Exception):
+    def __init__(self, actual_filetype: str, expected_filetype: str, message: str = ""):
+        self.actual_filetype = actual_filetype
+        self.expected_filetype = expected_filetype
+        if not message:
+            message = (
+                f"Extracted media type `{self.actual_filetype}` does not match "
+                f"expected media type `{self.expected_filetype}`."
+            )
+        super().__init__(message)
+
+
+def extract_filetype(url: str) -> tuple[str | None, str | None]:
```

Comment:
Is the return type more correctly put as `tuple[str, str] | tuple[None, None]`? It's hard to tell from the code, but it doesn't look like there are any cases where it would be mixed `str, None` or `None, str`, but this signature would include those possibilities.

Reply:
That seems correct though I kind of prefer the shorter general signature still.